test/functional/client-ca.test.ts

```diff
@@ -13,8 +13,104 @@
 //   '../fixtures/certs/server/fullchain.pem',
 // );
 
+import * as fs from 'fs';
+// import * as https from 'https';
+import * as path from 'path';
+// import axios from 'axios';
+import { BrokerClient, createBrokerClient } from '../setup/broker-client';
+import { BrokerServer, createBrokerServer } from '../setup/broker-server';
+import { createTestLogger } from '../helpers/logger';
+import { TestWebServer, createTestWebServer } from '../setup/test-web-server';
+import { requestAsync } from '../utils';
+
+const LOG = createTestLogger();
+const fixtures = path.resolve(__dirname, '..', 'fixtures');
+const serverAccept = path.join(fixtures, 'server', 'filters.json');
+const clientAccept = path.join(fixtures, 'client', 'filters-https.json');
+const sslCertificateKey = path.join(fixtures, 'certs/server/privkey.pem');
+const sslCertificate = path.resolve(fixtures, 'certs/server/fullchain.pem');
+
 describe('correctly use supplied CA cert on client for connections', () => {
-  it.skip('get an error trying to connect to a server with unknown CA', async () => {});
+  let tws: TestWebServer;
+  let bs: BrokerServer;
+  let bc: BrokerClient;
+  let brokerToken: string;
+
+  beforeAll(async () => {
+    process.env['NODE_TLS_REJECT_UNAUTHORIZED'] = '0';
+    // check if all fixtures files exist
+    try {
+      fs.readFileSync(sslCertificate);
+      fs.readFileSync(sslCertificateKey);
+      fs.readFileSync(serverAccept);
+      fs.readFileSync(clientAccept);
+    } catch (error) {
+      LOG.error(error);
+      throw new Error('check paths to fixtures');
+    }
+
+    tws = await createTestWebServer({
+      sslCertificatePath: sslCertificate,
+      sslCertificateKeyPath: sslCertificateKey,
+    });
+
+    bs = await createBrokerServer({ filters: serverAccept });
+
+    bc = await createBrokerClient({
+      brokerServerUrl: `http://localhost:${bs.port}`,
+      brokerToken: '12345',
+      filters: clientAccept,
+    });
+
+    await new Promise((resolve) => {
+      bs.server.io.on('connection', (socket) => {
+        socket.on('identify', (clientData) => {
+          brokerToken = clientData.token;
+          resolve(brokerToken);
+        });
+      });
+    });
+  });
+  afterAll(async () => {
+    await tws.server.close();
+    setTimeout(async () => {
+      await bc.client.close();
+    }, 100);
+    await new Promise<void>((resolve) => {
+      bc.client.io.on('close', () => {
+        resolve();
+      });
+    });
+
+    setTimeout(async () => {
+      await bs.server.close();
+    }, 100);
+    await new Promise<void>((resolve) => {
+      bs.server.io.on('close', () => {
+        resolve();
+      });
+    });
+  });
+
+  it.skip('get an error trying to connect to a server with unknown CA', async () => {
```

Comment:
Still skipping?

Reply:
I take out this file from the PR (request package throws an exception and exits).
Will tackle it with the next PR. Done with c3e8056.